### PR TITLE
Move `follow_redirects` parameter

### LINKED ISSUE #82335: Add parameter `follow_redirects` to the list of `url_argument_spec`
### Summary

Modules from external collections like [`community.general.rundeck_project`](https://github.com/ansible-collections/community.general/blob/1b9d437be8d085b714f2c38208b30401e7dcf2d9/plugins/module_utils/rundeck.py#L21) use the [`module_utils.url_argument_spec`](https://github.com/ansible/ansible/blob/3a42a0036875c8cab6a62ab9ea67a365e1dd4781/lib/ansible/module_utils/urls.py#L1101) function to extend their parameters with the ones Ansible provides. They use the implementation behind [`ansible.builtin.uri`] so that makes sense.

This feature request is about adding the missing parameter [`follow_redirects`](https://github.com/ansible/ansible/blob/3a42a0036875c8cab6a62ab9ea67a365e1dd4781/lib/ansible/modules/uri.py#L611).

Adding this parameter would allow to use modules that build upon the ansible-uri implementation to follow redirects as well, e.g. when the service they're trying to reach is located behind a reverse proxy.

Reference:
* This forum post: https://forum.ansible.com/t/reuse-code-from-modules-in-other-collections/2699
* This issue: https://github.com/ansible-collections/community.general/issues/7670

### Issue Type

Feature Idea

### Component Name

uri

### Additional Information

-

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by monsdar**
##### SUMMARY

Moves the `follow_redirects` parameter to the url_argument_spec function so external modules like community.general.rundeck_project could use it as well without having to copy it.

Fixes #82335 

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: lib/ansible/plugins/doc_fragments/url_redirect.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (felixfontein) [inline lib/ansible/plugins/doc_fragments/url_redirect.py]: I would personally not add a new docs fragment, but add a sub-docs fragment to the `url` docs fragment. Not sure what the Core team prefers though.

File: lib/ansible/plugins/doc_fragments/url_redirect.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class ModuleDocFragment(object):
+
+    # Standard files documentation fragment
+    DOCUMENTATION = r'''
+options:
+  follow_redirects:
+    description:
+      - Whether or not the URI module should follow redirects. V(all) will follow all redirects.
```
Comment (felixfontein) [inline lib/ansible/plugins/doc_fragments/url_redirect.py]: One problem is that this docs fragment now explicitly mentions the `uri` module. I would reformulate this to
```suggestion
      - Whether or not redirects should be followed. V(all) will follow all redirects.
```

**Review by felixfontein — COMMENTED** [review]
You definitely also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment).

Also please note that the `url` lookup also has `follow_redirects`, although with a different description, a different default, a `version_added`, etc. Here you can use the ability to overwrite docs fragments, by including the docs fragment, and specifying
```yaml
  follow_redirects:
    description: String of urllib2, all/yes, safe, none to determine how redirects are followed
    version_added: "2.10"
    default: 'urllib2'
    vars:
        - name: ansible_lookup_url_follow_redirects
    env:
        - name: ANSIBLE_LOOKUP_URL_FOLLOW_REDIRECTS
    ini:
        - section: url_lookup
          key: follow_redirects
```

File: lib/ansible/plugins/doc_fragments/url_redirect.py (hunk 1 above)
Comment (monsdar) [reply lib/ansible/plugins/doc_fragments/url_redirect.py]: I cannot find any docs on this, but it sounds like the way to go. Could you give me a pointer on how to implement it?

Comment (felixfontein) [reply lib/ansible/plugins/doc_fragments/url_redirect.py]: See for example:
- https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/doc_fragments/action_common_attributes.py#L53
- https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/import_playbook.py#L26

Comment (monsdar) [reply lib/ansible/plugins/doc_fragments/url_redirect.py]: I added the sub-doc fragment. I'm unsure about the uri-lookup docs though. The default there is `urllib2` while the uri-module has `safe` as default. Would this be overwritten in a wrong manner?

**Comment by monsdar** [comment]
Sorry for the long pause, Christmas got in the way 👀

File: lib/ansible/module_utils/urls.py [hunk 3]
```diff
@@ -1118,6 +1118,15 @@ def url_argument_spec():
     )
 
 
+def url_redirect_argument_spec():
+    '''
+    As an addition to `url_argument_spec` this spec contains a `follow_redirects` argument
+    '''
+    return dict(
+        follow_redirects=dict(type='str', default='safe', choices=['all', 'no', 'none', 'safe', 'urllib2', 'yes']),
```
Comment (Akasurde) [inline lib/ansible/module_utils/urls.py:1126]: @monsdar Please take a look at https://github.com/ansible/ansible/pull/83169.

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
superseded by https://github.com/ansible/ansible/pull/84442

**Pull request closed without merging**